airflow/jobs.py

```diff
@@ -1432,34 +1432,6 @@ def _process_executor_events(self, simple_dag_bag, session=None):
                 "Executor reports %s.%s execution_date=%s as %s",
                 dag_id, task_id, execution_date, state
             )
-            if state == State.FAILED or state == State.SUCCESS:
-                qry = session.query(TI).filter(TI.dag_id == dag_id,
-                                               TI.task_id == task_id,
-                                               TI.execution_date == execution_date)
-                ti = qry.first()
-                if not ti:
-                    self.log.warning("TaskInstance %s went missing from the database", ti)
-                    continue
-
-                # TODO: should we fail RUNNING as well, as we do in Backfills?
-                if ti.state == State.QUEUED:
-                    msg = ("Executor reports task instance %s finished (%s) "
-                           "although the task says its %s. Was the task "
-                           "killed externally?".format(ti, state, ti.state))
-                    self.log.error(msg)
-                    try:
-                        simple_dag = simple_dag_bag.get_dag(dag_id)
-                        dagbag = models.DagBag(simple_dag.full_filepath)
-                        dag = dagbag.get_dag(dag_id)
-                        ti.task = dag.get_task(task_id)
-                        ti.handle_failure(msg)
-                    except Exception:
-                        self.log.error("Cannot load the dag bag to handle failure for %s"
-                                       ". Setting task to FAILED without callbacks or "
-                                       "retries. Do you have enough resources?", ti)
-                        ti.state = State.FAILED
-                        session.merge(ti)
-                        session.commit()
```

Comment:
This seems like a scary thing to do - ripping out a whole "feature" in airflow.  Allow me to explain myself:

*CHANGE SUMMARY:*

This block was added in the following PR:
https://github[dot]com/apache/airflow/pull/2715
which was written to address the following ticket:
https://issues.apache[dot]org/jira/browse/AIRFLOW-1641

(The above URLS are de-fanged to avoid programatic posting of this comment directly onto upstream's repo or jira.  While I believe everything I'm saying here needs to be documented for our records, I don't feel we gain anything by shouting our displeasure through a megaphone.  In the future our airflow repo will be private and we will not need to take this kind of measure.)

This code lives inside of SchedulerJob._process_executor_events() - this method's job is to retrieve the exit status of celery executors on the workers, and report on that exit status.

This block I am removing has the following effect:
"If the celery executor finished, but the state of the task instance is still pending, fail the task instance."

I need to remove this block because it is now intended for celery executors to exit without changing the state of the task instance if the hostname is set from a previous worker instance, and the heartbeat has not expired.

*CHANGE JUSTIFICATION:*

The following is a reasonably compact summary of the timeline of events limited to the pieces that were relevant to my decision to drop this feature rather than tweak its implementation:

- The author of the ticket (The Reporter) asserted that his tasks were getting stuck in a pending state because of a timeout loading the dag in `DagBag.process_file()` . This code runs on the worker prior to setting state to running.  If a timeout occurs, the worker swallows the error and the celery task returns successfully, but the task instance state isn't changed from pending, and the task is not run.
- The author of the pull request (The Implementer) put up a PR and requested that The Reporter test his changes because he (The Implementer) hadn't tested his own changes.  (I mention this because at no point in the ticket or the pull request does anybody assert that they have tested that this feature behaves as intended, and there are mistakes here that I believe would have been caught if someone had tested it, like the visibly incorrect log statement on 1447-1448.)
- The first two comments on this block of code in the PR from airflow maintainers are "Is there nothing better we can do here?" and "I share this concern."
- After addressing code read comments, The Implementer asserts that he is uncomfortable with the state of these changes because they introduce an inconsistency between the way BackfillJobs are handled when compared to other tasks.

In addition to the lack of confidence the above timeline inspires, I also feel strongly that this implementation is incorrect.  If a timeout importing dags occurs, the celery executor should fail, and that should trigger a task failure.

You'll notice reading this that I didn't go ahead and change this procedure to what I describe above as the correct implementation.  There are several reasons for this:

- The dag importing code is used many places in airflow, so removing the swallowing of dag import timeouts is a surprisingly invasive change.
- We do not have the problem of dags failing to import within the timeout, and I hesitate to invest much engineering time into fixing edge cases we do not have that only arise when airflow is abused.  (Dag generation should not be expensive.)
- Doing this correctly involves making a variety of changes to airflow that would be difficult to merge into our upcoming upstream patch.

After consideration of all of the above, I elected to simply remove this "feature" entirely from airflow.  I do not think we will miss it, and if we do fall prey to the issue "fixed" by this "feature", it should be a quick matter of inspecting the logs to see that this is what has happened.

*TL;DR:*
This code is a problem, so I'm removing it.  I think it's unlikely we'll regret this, and it's easy to reverse if we do.

Reply:
you are doing the Lord's work

Reply:
hmm... im blocked from those links

Reply:
Those links are de-fanged - you'll need to replace [dot] with .